### PR TITLE
Add checks for duplicate glossary texts

### PR DESCRIPTION
**Pull request by 73nici**

### CODE REVIEW

File: Classes/Domain/Repository/GlossaryEntryRepository.php [hunk 1]
```diff
@@ -48,6 +48,26 @@ public function findByLanguages(array $languages): array
         return $query->execute()->toArray();
     }
 
+    /**
+     * @param string $text
```
Comment (robert-heinig) [inline Classes/Domain/Repository/GlossaryEntryRepository.php]: Does not add value to the PHP Type, so could be omitted. (And describes only 1 of 2 arguments.)

File: Classes/Domain/Repository/GlossaryEntryRepository.php [hunk 2]
```diff
@@ -48,6 +48,26 @@ public function findByLanguages(array $languages): array
...
+     * @return GlossaryEntry[]
+     */
+    public function findByTextAndLanguage(string $text, string $language): array
+    {
+        $query = $this->createQuery();
+        $queryBuilder = $query->getQueryBuilder();
+
+        $result = $queryBuilder
+            ->select('e.glossaryLanguage')
```
Comment (robert-heinig) [inline Classes/Domain/Repository/GlossaryEntryRepository.php]: Could be count() to simplify the process and return a boolean directly?

File: Classes/Controller/GlossaryController.php [hunk 3]
```diff
@@ -111,6 +112,15 @@ public function createAction(): void
...
         $aggregateIdentifier = Algorithms::generateUUID();
 
+
+        foreach ($texts as $language => $text) {
+            if (!empty($this->glossaryEntryRepository->findByTextAndLanguage($text, $language))) {
+                throw new InvalidArgumentException(sprintf('Text "%s" already exists with language "%s".', $text, $language));
+            }
+        }
+
+        $this->addErrorFlashMessage();
```
Comment (robert-heinig) [inline Classes/Controller/GlossaryController.php]: So the Exception above gets automatically transformed to a flash message output?

**Review by robert-heinig — CHANGES_REQUESTED** [review]

File: Classes/Controller/GlossaryController.php (hunk 3 above)
Comment (73nici) [reply Classes/Controller/GlossaryController.php]: No, unfortunatly not, thats just a leftover from testing. I removed it 👍🏻

File: Classes/Domain/Repository/GlossaryEntryRepository.php (hunk 2 above)
Comment (73nici) [reply Classes/Domain/Repository/GlossaryEntryRepository.php]: I adjusted it to select with count. `count()` is not available in this querybuilder version.

File: Classes/Domain/Repository/GlossaryEntryRepository.php (hunk 1 above)
Comment (73nici) [reply Classes/Domain/Repository/GlossaryEntryRepository.php]: Removed the `@param` notation 👍🏻.

**Review by robert-heinig — APPROVED** [review]

**Review by robert-heinig — APPROVED** [review]

**Pull request merged**